From 6088d48f0f10907c327e4abfb290bef23919b2c5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Juan=20Manuel=20L=C3=B3pez=20Carrillo?=
 <juanmanuellopezcarrillo@gmail.com>
Date: Tue, 14 Jul 2026 01:26:53 +0200
Subject: [PATCH 063/143] arm64: dts: sun55i-a523: add the DDR THS

thermal-sensor@200a000 node (1 DDR sensor, clock CLK_GPADC0, shared reset
RST_BUS_THS) and the ddr-thermal zone (monitoring only, no trips, like the
BSP).
---
 arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi b/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
index 5b0a6396c..031bdabcf 100644
--- a/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
@@ -203,6 +203,12 @@ npu-thermal {
 			polling-delay = <1000>;
 			thermal-sensors = <&ths 3>;
 		};
+
+		ddr-thermal {
+			polling-delay-passive = <0>;
+			polling-delay = <1000>;
+			thermal-sensors = <&ths_ddr 0>;
+		};
 	};
 
 	soc {
@@ -535,6 +541,17 @@ ths: thermal-sensor@2009400 {
 			#thermal-sensor-cells = <1>;
 		};
 
+		ths_ddr: thermal-sensor@200a000 {
+			compatible = "allwinner,sun55i-a523-ths-ddr";
+			reg = <0x0200a000 0x400>;
+			clocks = <&ccu CLK_BUS_THS>, <&ccu CLK_GPADC0>;
+			clock-names = "bus", "mod";
+			resets = <&ccu RST_BUS_THS>;
+			nvmem-cells = <&ths_calibration>;
+			nvmem-cell-names = "calibration";
+			#thermal-sensor-cells = <1>;
+		};
+
 		iommu: iommu@2010000 {
 				compatible = "allwinner,sun55i-a523-iommu",
 				"allwinner,sun50i-h616-iommu";
-- 
2.53.0

